CMakeModules/CPackConfig.cmake

```diff
@@ -49,91 +88,170 @@ if(APPLE)
   set(CPACK_RESOURCE_FILE_LICENSE ${LICENSE_FILE_OUT})
   set(CPACK_RESOURCE_FILE_README ${README_FILE_OUT})
   set(CPACK_RESOURCE_FILE_WELCOME ${WELCOME_FILE_OUT})
+elseif(WIN32)
+  set(WIN_INSTALL_SOURCE ${PROJECT_SOURCE_DIR}/CMakeModules/nsis)
+
+  set(LICENSE_FILE       "${ArrayFire_SOURCE_DIR}/LICENSE")
+  set(LICENSE_FILE_OUT   "${CMAKE_CURRENT_BINARY_DIR}/license.txt")
+  configure_file(${LICENSE_FILE} ${LICENSE_FILE_OUT})
+  set(CPACK_RESOURCE_FILE_LICENSE ${LICENSE_FILE_OUT})
+
+  #NSIS SPECIFIC VARIABLES
+  set(CPACK_NSIS_ENABLE_UNINSTALL_BEFORE_INSTALL ON)
+  set(CPACK_NSIS_MODIFY_PATH ON)
+  set(CPACK_NSIS_DISPLAY_NAME "${APP_NAME}")
+  set(CPACK_NSIS_PACKAGE_NAME "${APP_NAME}")
+  set(CPACK_NSIS_HELP_LINK "${APP_URL}")
+  set(CPACK_NSIS_URL_INFO_ABOUT "${APP_URL}")
+  set(CPACK_NSIS_INSTALLED_ICON_NAME "${MY_CPACK_PACKAGE_ICON}")
+  if (CMAKE_CL_64)
+    set(CPACK_NSIS_INSTALL_ROOT "$PROGRAMFILES64")
+  else (CMAKE_CL_64)
+    set(CPACK_NSIS_INSTALL_ROOT "$PROGRAMFILES")
+  endif (CMAKE_CL_64)
 else()
   set(CPACK_RESOURCE_FILE_LICENSE "${ArrayFire_SOURCE_DIR}/LICENSE")
   set(CPACK_RESOURCE_FILE_README "${ArrayFire_SOURCE_DIR}/README.md")
 endif()
 
-# Long description of the package
-set(CPACK_PACKAGE_DESCRIPTION
-"ArrayFire is a high performance software library for parallel computing
-with an easy-to-use API. Its array based function set makes parallel
-programming simple.
+# Set the default components installed in the package
+get_cmake_property(CPACK_COMPONENTS_ALL COMPONENTS)
 
-ArrayFire's multiple backends (CUDA, OpenCL and native CPU) make it
-platform independent and highly portable.
+include(CPackComponent)
 
-A few lines of code in ArrayFire can replace dozens of lines of parallel
-computing code, saving you valuable time and lowering development costs.")
+cpack_add_install_type(Development
+  DISPLAY_NAME "Development")
+cpack_add_install_type(Extra
+  DISPLAY_NAME "Extra")
+cpack_add_install_type(Runtime
+  DISPLAY_NAME "Runtime")
 
-# Short description of the package
-set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "A high performance library for parallel computing with an easy-to-use API.")
+cpack_add_component_group(dependencies
+                          DISPLAY_NAME "Dependencies"
+                          DESCRIPTION "ArrayFire dependencies"
+                          EXPANDED BOLD_TITLE)
 
-# Set the default components installed in the package
-set(CPACK_COMPONENTS_ALL cpu cuda opencl unified headers documentation cmake examples)
+cpack_add_component(mkl_dependencies
```

Comment:
I think MKL dependencies shouldn't be optional because they are not usually available via package managers or cuda toolkits

Reply:
Not sure what you mean ? I didn't make them optional intentionally, why do you think they are an optional component now ?